src/server/db_slice.cc

```diff
@@ -221,18 +207,18 @@ unsigned PrimeEvictionPolicy::Evict(const PrimeTable::HotspotBuckets& eb, PrimeT
 // because some other transaction might conclude and clear the fetched_items_ with OnCbFinish()
 class FetchedItemsRestorer {
  public:
-  using RestoreType = absl::flat_hash_set<CompactObjectView>;
-  explicit FetchedItemsRestorer(RestoreType* dst) : dst_to_restore_(dst) {
-    cached_ = std::move(*dst_to_restore_);
+  // using RestoreType = absl::flat_hash_set<CompactObjectView>;
```

Comment:
this code should be deleted. hacking it as a noop to get the initial feedback for review. @adiholden  @chakaz  FYI

Reply:
As far as I understand this code comes to protect from DbSlice::OnCbFinish calling fetched_items_.clear to not clear the fetched items in caller transaction while the preempt
OnCbFinish is called on each transaction finish so if we have 2 interliving transactions one running the OnCbFinish after a single key was fetched before another one was fetched it will clear the vector not as expected

Reply:
Have you read the PR description? My claim is that if a tx preempts we do not need to protect iterators - and that was the purpose of `fetched_items_`

Reply:
so why did you keep fetched_items_ logic and moved to finger print ? if this is the claim we can remove fetched_items_ at all